R/image-interact.R

```diff
@@ -281,8 +278,8 @@ nearPoints <- function(df, coordinfo, xvar = NULL, yvar = NULL,
     stop("nearPoints: `yvar` ('", yvar ,"')  not in names of input")
 
   # Extract data values from the data frame
-  x <- asNumber(df[[xvar]])
-  y <- asNumber(df[[yvar]])
+  x <- asNumber(df[[xvar]], coordinfo$domain$discrete_limits$x)
```

Comment:
Make sure to account for `NA`s here.

Reply:
`NA`s aren't an issue here (in `nearPoints()`) because we end up [using `which()`](https://github.com/rstudio/shiny/blob/6fed1c60ac018c84afd1b8aeba54f5a70ebbb37c/R/image-interact.R#L314) and subsetting by that resulting integer vector rather than a logical index